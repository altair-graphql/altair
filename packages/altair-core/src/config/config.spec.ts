import { describe, expect, it } from '@jest/globals';
import { AltairConfig } from '.';

describe('config', () => {
  it('creates config object with expected properties', () => {
    const config = new AltairConfig();

    expect(config).toMatchSnapshot();
  });
});
