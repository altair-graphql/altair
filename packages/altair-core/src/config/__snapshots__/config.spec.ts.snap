// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config creates config object with expected properties 1`] = `
AltairConfig {
  "add_query_depth_limit": 3,
  "defaultTheme": "system",
  "default_language": "en-US",
  "disableLineNumbers": false,
  "donation": {
    "action_count_threshold": 50,
    "url": "https://opencollective.com/altair/donate",
  },
  "ga": "UA-41432833-6",
  "initialData": {
    "disableAccount": false,
    "environments": {},
    "headers": "",
    "initialHttpMethod": "POST",
    "initialRequestHandlerAdditionalParams": {},
    "initialRequestHandlerId": "http",
    "initialSubscriptionRequestHandlerId": "websocket",
    "initialSubscriptionsPayload": {},
    "instanceStorageNamespace": "altair_",
    "persistedSettings": undefined,
    "postRequestScript": "",
    "preRequestScript": "",
    "preserveState": true,
    "query": "",
    "settings": undefined,
    "subscriptionsEndpoint": "",
    "subscriptionsProtocol": "",
    "url": "",
    "variables": "",
    "windows": [],
  },
  "isTranslateMode": undefined,
  "isWebApp": undefined,
  "languages": {
    "cs-CZ": "Czech",
    "de-DE": "German",
    "en-US": "English",
    "es-ES": "Espa√±ol",
    "fr-FR": "French",
    "it-IT": "Italian",
    "ja-JP": "Japanese",
    "ko-KR": "Korean",
    "pl-PL": "Polish",
    "pt-BR": "Brazilian",
    "ro-RO": "Romanian",
    "ru-RU": "Russian",
    "sr-SP": "Serbian",
    "uk-UA": "Ukrainian",
    "vi-VN": "Vietnamese",
    "zh-CN": "Chinese Simplified",
  },
  "max_windows": 15,
  "query_history_depth": 15,
  "tab_size": 2,
  "themes": [
    "light",
    "dark",
    "dracula",
    "system",
  ],
  "useLocalSandboxUrl": false,
}
`;
